tests/foreman/cli/test_classparameters.py

```diff
@@ -948,55 +1292,43 @@ def test_negative_create_matcher_avoid_duplicate(self):
         """
 
     @run_only_on('sat')
-    @stubbed()
-    @tier2
+    @tier1
     def test_positive_remove_matcher(self):
         """Removal of matcher from parameter.
 
         @id: f51ea9ca-f57c-482e-841f-3ea5cc8f8958
 
         @steps:
 
-        1. Override the parameter and create a matcher
-        for some attribute.
+        1. Override the parameter and create a matcher for some attribute.
         2. Remove the matcher created in step 1.
 
         @assert: The matcher removed from parameter.
-
-        @caseautomation: notautomated
-
-        @CaseLevel: Integration
         """
+        sc_param_id = self.sc_params_ids_list.pop()
+        value = gen_string('alpha')
+        SmartClassParameter.add_override_value({
+            'smart-class-parameter-id': sc_param_id,
+            'match': 'is_virtual=true',
+            'value': value
+        })
+        sc_param = SmartClassParameter.info({
+            'puppet-class': 'ntp',
+            'id': sc_param_id,
+        })
+        self.assertEqual(len(sc_param['override-values']['values']), 1)
+        SmartClassParameter.remove_override_value({
+            'smart-class-parameter-id': sc_param_id,
+            'id': sc_param['override-values']['values']['1']['id'],
+        })
+        sc_param = SmartClassParameter.info({
+            'puppet-class': 'ntp',
+            'id': sc_param_id,
+        })
+        self.assertEqual(len(sc_param['override-values']['values']), 0)
 
     @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_positive_impact_parameter_delete_attribute(self):
```

Comment:
CLI does not operate with thing like attribute. You do not need to create new attribute like 'is_virtual', to create new matcher `is_virtual=true` like we do in UI. Here you just write `is_virtual=true` and get the result, so it is impossible to delete something which was not created